contrib/awslambda/python/src/python/pants/contrib/awslambda/python/tasks/lambdex_run.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import functools
+import os
+import shutil
+
+from pants.base.build_environment import get_buildroot
+from pants.base.exceptions import TaskError
+from pants.base.workunit import WorkUnitLabel
+from pants.task.task import Task
+from pants.util.contextutil import temporary_dir
+from pants.util.dirutil import safe_mkdir_for
+from pants.util.fileutil import atomic_copy
+
+from pants.contrib.awslambda.python.targets.python_awslambda import PythonAWSLambda
+from pants.contrib.awslambda.python.tasks.lambdex_prep import LambdexPrep
+
+
+class LambdexRun(Task):
+  """Runs Lambdex to convert pexes to AWS Lambda functions.
+
+  Note that your pex must be built to run on Amazon Linux, e.g., if it contains native code.
+  When deploying the lambda, its handler should be set to `lambdex_handler.handler`, which
+  is a wrapper around the target-specified handler.
+  """
+
+  @staticmethod
+  def _is_python_lambda(target):
+    return isinstance(target, PythonAWSLambda)
+
+  @classmethod
+  def product_types(cls):
+    return ['python_aws_lambda']
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(LambdexRun, cls).prepare(options, round_manager)
+    round_manager.require_data(LambdexPrep.tool_instance_cls)
+    round_manager.require('pex_archives')
+
+  @classmethod
+  def create_target_dirs(self):
+    return True
+
+  def execute(self):
+    targets = self.get_targets(self._is_python_lambda)
+    with self.invalidated(targets=targets) as invalidation_check:
```

Comment:
iirc `invalidated` is a little expensive even with empty targets, so would recommend adding
```
if not targets:
  return
```

Reply:
I'm not aware of such a performance issue, but if so, this check should be inside `invalidated`, probably, rather than every task having to think of it.